Automattic/wp-calypso
# Purchases: Remove PurchaseCardDetails inheritance

Remove `PurchaseCardDetails`, which subclasses `React.Component`, but is not a renderable `Component` instead is intended for inheritance.

[React docs discourage inheritance](https://reactjs.org/docs/composition-vs-inheritance.html#so-what-about-inheritance) and it is not used as a convention in Calypso components. Indeed, the use of this class obscured prop usage and only saved a bit of code duplication.

This PR takes the simple approach of duplicating a few methods and removing the intermediate class.

An alternative would be to add a specific component containing `CreditCardForm`.

## Testing
1. [(/me/purchases/)](https://calypso.live/me/purchases?branch=remove/purchasecarddetails-superclass&debug)
1. `localStorage.debug = "calypso:analytics:tracks"`
1. Pick a purchase (purchased with credits is a good idea).
1. Add a credit card. Does the form work as expected?
1. Edit the credit card. Does it continue to work well?
1. Was the `calypso_purchases_credit_card_form_submit` event fired on form submission?

Prep for #25219
